snippets/kubernetes.json

```diff
@@ -460,5 +460,834 @@
+            "  selector:",
+            "    matchLabels:",
+            "      $3",
+            "  template:",
+            "    metadata:",
+            "      labels:",
+            "        ${3:app: $1}",
+            "    spec:",
+            "      containers:",
+            "      - name: ${4:name}",
+            "        image: ${5:nginx}",
+            "        resources:",
+            "          requests:",
+            "            cpu: 100m",
+            "            memory: 200Mi",
+            "        ports:",
+            "        - containerPort: 8080"
+        ],
+        "prefix": "dep"
+    },
+    "Init Container": {
+        "body": [
+            "initContainers:",
+            "- name: ${1:myinit}",
+            "  image: ${2:busybox}",
+            "  command: [${3:rm, -rf,  $5/lost+found}]",
+            "  ${6:volumeMounts:",
+            "  - name: ${4:data}",
+            "    mountPath: ${5:/data}}$0"
+        ],
+        "prefix": "init"
+    },
+    "tls-acme annotations": {
+        "body": [
+            "annotations:",
+            "  kubernetes.io/tls-acme: \"true\""
+        ],
+        "prefix": "atls"
+    },
+    "Selector MatchLabels": {
+        "body": [
+            "selector:",
+            "\tmatchLabels:",
+            "\t\t$0"
+        ],
+        "prefix": "selm"
+    },
+    "Role and Binding": {
+        "body": [
+            "kind: ${1:Cluster}Role",
+            "apiVersion: rbac.authorization.k8s.io/v1",
+            "metadata:",
+            "  ${2:namespace: ${3:default}",
+            "  }name: ${4:configmap-updater}",
+            "rules:",
+            "- apiGroups: [\"${5:}\"]",
+            "  resources: [\"${6:configmaps}\"]",
+            "  resourceNames: [\"${7:my-configmap}\"]",
+            "  verbs: [${8:\"update\", \"get\"}]",
+            "---",
+            "# This role binding allows \"jane\" to read pods in the \"default\" namespace.",
+            "kind: ${1:Cluster}RoleBinding",
+            "apiVersion: rbac.authorization.k8s.io/v1",
+            "metadata:",
+            "  name: $4",
+            "  $2",
+            "subjects:",
+            "- kind: ${9:User|ServiceAccount|Group}",
+            "  name: ${10:jane} # Name is case sensitive",
+            "  ${2:namespace: ${3:default}",
+            "}roleRef:",
+            "  kind: $1Role #this must be Role or ClusterRole",
+            "  name: $4 # this must match the name of the Role or ClusterRole you wish to bind to",
+            "  apiGroup: rbac.authorization.k8s.io"
+        ],
+        "prefix": "rbac"
+    },
+    "StatefulSet": {
+        "body": [
+            "apiVersion: v1",
+            "kind: Service",
+            "metadata:",
+            "  name: ${1:myservice}",
+            "spec:",
+            "  ports:",
+            "  - port: $5",
+            "    name: $6",
+            "  clusterIP: None",
+            "  selector:",
+            "    $2",
+            "---",
```

Comment:
Lines 1235 - 1246 repeat what you included in the service snippet, and then most of the remaining snippet is already included via the `k-statefulset` snippet 

However, if you rename this to be something like `k-statefulset-volumeclaim` we should be ok including it.